platform/darwin/src/MGLGeometry_Private.h

```diff
@@ -13,6 +13,18 @@
 CGRect MGLExtendRect(CGRect rect, CGPoint point);
 
 NS_INLINE mbgl::LatLng MGLLatLngFromLocationCoordinate2D(CLLocationCoordinate2D coordinate) {
+    if (std::isnan(coordinate.latitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"latitude must not be NaN"];
+    }
+    if (std::isnan(coordinate.longitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"longitude must not be NaN"];
+    }
+    if (std::abs(coordinate.latitude) > 90.0) {
```

Comment:
I think we’ll want to push these checks further up. In many places where we call this function, the analogous MapKit API fails silently when given out-of-bounds coordinates; an exception would be heavy-handed. In other cases, a debug assertion is all that’s needed.